carma_wm_ctrl/src/GeofenceSchedule.cpp

```diff
@@ -14,92 +14,111 @@
  * the License.
  */
 
-#include <ros/time.h>
-#include <carma_wm_ctrl/GeofenceSchedule.h>
+#include <carma_wm_ctrl/GeofenceSchedule.hpp>
 #include <boost/date_time/posix_time/posix_time.hpp>
-#include <ros/ros.h>
 namespace carma_wm_ctrl
 {
+
+double toSec(const boost::posix_time::time_duration& duration) {
+  if (duration.is_special()) {
+    throw std::invalid_argument("Cannot convert special duration to seconds");
+  }
+  return duration.total_microseconds() / 1000000.0;
+}
+
+double toSec(const boost::posix_time::ptime& time) {
```

Comment:
These look like they were copied and pasted from the methods in lanelet2_extension. Why not just include the timing header from that library and call them directly?

Reply:
refactored the time namespace into different file and included it here